src/app/components/Item/ItemTableRow.jsx

```diff
@@ -44,11 +46,26 @@ class ItemTableRow extends React.Component {
 
   aeonUrl(item) {
     const url = Array.isArray(item.aeonUrl) ? item.aeonUrl[0] : item.aeonUrl;
-    const barcodeElement = `${item.barcode ? `&barcode=${item.barcode}` : ''}`;
-    const idElement = `${!url.includes('itemid') && item.id ? `&itemid=${item.id}` : ''}`;
-    const shelfMarkElement = `${item.callNumber ? `&shelfMark=${item.callNumber}` : ''}`;
+    const searchParams = new URL(url).searchParams;
+    const paramMappings = {
+      ItemISxN: 'id',
+      itemNumber: 'barcode',
+      CallNumber: 'callNumber',
+    };
+
+    let params = Object.keys(paramMappings).map((paramName) => {
+      if (searchParams.has(paramName)) return null;
```

Comment:
I've not found a bib with multiple Aeon requestable items, so I assume that whatever params are in hte original 856 will be accurate for the sole item. So, yes, it seems fine to leave the param unchanged when it already exists in the URL.